src/simulation.jl

```diff
@@ -261,19 +261,27 @@ function get_vars_code(d::AbstractNamedDecapode, vars::Vector{Symbol}, ::Type{st
   stmts = quote end
   
   map(vars) do s
-    # If name is not unique then error
+    # If name is not unique (or not just literals) then error
     found_names_idxs = incident(d, s, :name)
     # TODO: we should handle the case of same literals better
-    if length(found_names_idxs) > 1 && any(d[found_names_idxs, :type] .!== :Literal)
+    is_singular = length(found_names_idxs) == 1
+    is_all_literals = all(d[found_names_idxs, :type] .== :Literal)
+
+    if !is_singular && !is_all_literals
       throw(AmbiguousNameException(s, found_names_idxs))
     end
 
+    if is_all_literals
+      push!(stmts.args, :($s = $(parse(stateeltype, String(s)))))
```

Comment:
It looks like this is still creating assignments for literals.

Reply:
Yeah that's right. I just forgot to move the comment up with it.